proxy/operator_resource/op_eth_signing_api.py

```diff
@@ -0,0 +1,82 @@
+from typing import ClassVar
+
+from common.ethereum.errors import EthError
+from common.ethereum.hash import EthTxHash, EthAddress
+from common.jsonrpc.errors import InvalidParamError
+from common.neon.account import NeonAccount
+from common.utils.format import hex_to_bytes
+from common.utils.json_logger import logging_context
+from .server_abc import OpResourceApi
+from ..base.op_api import OpSignEthMessageRequest, OpSignEthMessageResp, OpSignEthTxRequest, OpSignEthTxResp
+
+
+class OpEthSigningApi(OpResourceApi):
+    name: ClassVar[str] = "OpResource::EthSigning"
+
+    @OpResourceApi.method(name="signEthMessage")
+    async def sign_eth_message(self, request: OpSignEthMessageRequest) -> OpSignEthMessageResp:
+        with logging_context(ctx=request.ctx_id):
+            if not self._cfg.enable_private_api:
+                raise EthError(message="private api is disabled")
+
+            try:
+                data = hex_to_bytes(request.data)
+                message = str.encode(f"\x19Ethereum Signed Message:\n{len(data)}") + data
+                neon_account = await self._neon_account(request.eth_address)
+                signed_message = neon_account.sign_msg(message)
+                return OpSignEthMessageResp(signed_message=signed_message.to_hex())
+            except Exception as exc:
+                _LOG.error("Failed to sign message", exc_info=exc)
+                raise InvalidParamError(message="bad input data")
+
+    @OpResourceApi.method(name="signEthTransaction")
+    async def sign_eth_transaction(self, request: OpSignEthTxRequest) -> OpSignEthTxResp:
+        with logging_context(ctx=request.ctx_id, chain_id=request.chain_id):
+            if not self._cfg.enable_private_api:
+                raise EthError(message="private api is disabled")
+
+            try:
+                if request.tx.from_address.is_empty:
+                    raise InvalidParamError(message="no sender in transaction")
+
+                # We use "data" here because eth-account hasn't been updated
+                # yet, but the correct field is "input"
+                # https://github.com/ethereum/go-ethereum/pull/28078
+
+                tx = {
+                    "to": request.tx.to_address.to_checksum(""),
+                    "chainId": request.chain_id,
+                    "nonce": request.tx.nonce,
+                    "value": request.tx.value,
+                    "data": request.tx.call_data.to_string(""),
+                    "gas": request.tx.gas_limit,
+                    "gasPrice": request.tx.gas_price,
+                }
+
+                neon_account = await self._neon_account(request.tx.from_address)
```

Comment:
It should be an account from local storage.
Not an operator account.
This part of the API is the implementation of:
https://geth.ethereum.org/docs/fundamentals/account-management

Reply:
There is now a separate "from" address not included within the request transaction. This from will be compared to the secret manager.

I think this is what you mean.